crates/storage/src/storage_provider.rs

```diff
@@ -2,26 +2,30 @@ use axum::async_trait;
 use std::path::Path;
 use tokio::fs::File;
 
-use crate::storage_error::StorageError;
-
 #[async_trait]
 pub trait StorageProvider: Send + Sync {
-    async fn open_file(&self, file_path: &Path) -> Result<File, StorageError>;
-    async fn open_or_create_file(&self, file_path: &Path) -> Result<File, StorageError>;
-    async fn read_file(&self, file: &mut File) -> Result<String, StorageError>;
-    async fn read(&self, path: &Path) -> Result<Vec<u8>, StorageError>;
+    type Err;
+    type StoredFile;
```

Comment:
I don't think this will work w/o any traits -- how will consuming code know how to modify the file?